12-offer-encoding.md

```diff
@@ -0,0 +1,1053 @@
+  - MUST NOT set or imply any `chain_hash` not set or implied by the offer.
+  - MUST set `payer_signature` `sig` as detailed in [Signature Calculation](#signature-calculation) using the `payer_key`.
+  - if the offer had a `quantity_min` or `quantity_max` field:
+    - MUST set `quantity`
+    - MUST set it within that (inclusive) range.
+  - otherwise:
+    - MUST NOT set `quantity`
+  - if the offer did not specify `amount`:
+    - MUST specify `amount`.`msat` in multiples of the minimum lightning-payable unit
+      (e.g. milli-satoshis for bitcoin) for the first `chains` entry.
+  - otherwise:
+    - MAY omit `amount`.
+    - if it sets `amount`:
+      - MUST specify `amount`.`msat` as greater or equal to amount expected by the offer
+        (before any proportional period amount).
+  - if the sender has a previous unpaid invoice (for the same offer) which it wants to cancel:
+    - MUST set `payer_key` to the same as the previous invoice.
+    - MUST set `replace_invoice` to the `payment_hash` or the previous invoice.
+  - if the offer contained `recurrence`:
+    - for the initial request:
+      - MUST use a unique `payer_key`.
+      - MUST set `recurrence_counter` `counter` to 0.
+    - for any successive requests:
+      - MUST use the same `payer_key` as the initial request.
+      - MUST set `recurrence_counter` `counter` to one greater than the highest-paid invoice.
+    - if the offer contained `recurrence_base` with `start_any_period` non-zero:
+      - MUST include `recurrence_start`
+      - MUST set `period_offset` to the period the sender wants for the initial request
+      - MUST set `period_offset` to the same value on all following requests.
+    - otherwise:
+      - MUST NOT include `recurrence_start`
+    - MAY set `payer_info` to arbitrary data to be reflected into the invoice.
+    - MAY set `payer_note` to arbitrary string to be reflected into the invoice.
+    - if the offer contained `recurrence_limit`:
+      - MUST NOT send an `invoice_request` for a period greater than `max_period`
+    - SHOULD NOT send an `invoice_request` for a period which has
+      already passed.
+    - if the offer contains `recurrence_paywindow`:
+      - if the offer has a `recurrence_basetime` or the `recurrence_counter` is non-zero:
+        - SHOULD NOT send an `invoice_request` for a period prior to `seconds_before` seconds before that period start.
+        - SHOULD NOT send an `invoice_request` for a period later than `seconds_after` seconds past that period start.
+    - otherwise:
+      - SHOULD NOT send an `invoice_request` with `recurrence_counter`
+        is non-zero for a period whose immediate predecessor has not
+        yet begun.
+  - otherwise:
+    - MUST NOT set `recurrence_counter`.
+    - MUST NOT set `recurrence_start`
+
+The reader of an invoice_request:
+  - MUST fail the request if `payer_key` is not present.
+  - MUST fail the request if `chains` does not include (or imply) a supported chain.
```

Comment:
no chain means "bitcoin mainnet", as specified in the writer side.

Reply:
just a suggestion because it wasn't clear to me, even though now I remember "no chain defaults to bitcoin mainchain" is said somewhere else in the spec. So unlike me probably most people will always remember that while reading.
How about...
```
  - MUST fail the request if `chains` does not include a supported chain (an empty chains field defaults to bitcoin mainchain)
 ```

or something of the short?

* Just saying, personally, what "(or imply)" in the middle of an already very abstract (as it should be) specification document can be compared to what some cats feel when a pink flower is carefully placed on top of their foreheads, gifs say. Sorry, the last part doesn't belong here. Sorry.